app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt

```diff
@@ -0,0 +1,251 @@
+import com.github.se.project.ui.navigation.NavigationActions
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.spy
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+class TutorEndToEndTest {
+
+  @Mock lateinit var navigationActions: NavigationActions
+
+  @Mock lateinit var context: Context
+
+  // Mock du ProfilesRepository
+  private val mockProfileRepository = mock(ProfilesRepository::class.java)
+
+  private val mockProfileViewModel = ListProfilesViewModel(mockProfileRepository)
+
+  private val mockLessonRepository = mock(LessonRepository::class.java)
+
+  private var mockLessonViewModel = spy(LessonViewModel(mockLessonRepository))
+
+  private var mockLessons =
+      listOf(
+          Lesson(
+              id = "1",
+              title = "Physics Tutoring",
+              description = "Mechanics and Thermodynamics",
+              subject = Subject.PHYSICS,
+              languages = listOf(Language.ENGLISH),
+              tutorUid = listOf(),
+              studentUid = "student123",
+              minPrice = 20.0,
+              maxPrice = 50.0,
+              timeSlot = "16/11/2024T12:00:00",
+              status = LessonStatus.STUDENT_REQUESTED,
+              latitude = 46.518973490411526,
+              longitude = 6.5685102716088295),
+      )
+
+  private val mockStudent =
+      Profile(
+          "student123",
+          "mockTutor",
+          "Ozymandias",
+          "Halifax",
+          "1234567890",
+          Role.STUDENT,
+          Section.IN,
+          AcademicLevel.BA3,
+          listOf(Language.ENGLISH),
+          listOf(Subject.AICC),
+          List(7) { List(12) { 0 } })
+
+  @get:Rule val composeTestRule = createComposeRule()
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getProfiles(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+      onSuccess(listOf(mockStudent)) // Simulate a list of profiles with our beloved Ozymandias
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
```

Comment:
This implementation of the mock means only 1 lesson can exist at a time, but I suppose having more isn't really necessary for testing, so this works.